plan/plan.go

```diff
@@ -195,20 +196,27 @@ type PhysicalPlan interface {
 	// ExplainInfo returns operator information to be explained.
 	ExplainInfo() string
 
-	// getChildrenPossibleProps tries to push the required properties to its children and return all the possible properties.
-	getChildrenPossibleProps(prop *requiredProp) [][]*requiredProp
+	// getChildReqProps gets the required property by child index.
+	getChildReqProps(idx int) *requiredProp
 
 	// statsProfile will return the stats for this plan.
 	statsProfile() *statsProfile
 }
 
 type baseLogicalPlan struct {
 	basePlan *basePlan
-	taskMap  map[string]task
+
+	taskMap map[string]task
 }
 
 type basePhysicalPlan struct {
 	basePlan *basePlan
+
+	reqProps []*requiredProp
```

Comment:
add comment. or `childrenReqProps` is better?

Reply:
fixed